### PR TITLE
Add `EventHandler` trait to replace `EventFn`

### PR DESCRIPTION
**Pull request by a1phyr**
Replace `EventFn`trait with `EventHandler`.

The new trait has new capabilities but can be implemented for more types, such as channel receivers which are very commonly used.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -139,9 +139,31 @@ mod config;
 mod error;
 
 /// The set of requirements for watcher event handling functions.
```
Comment (0xpr03) [inline src/lib.rs:141]: Can you add an example in the documentation for using the new impls ?

**Review by 0xpr03 — CHANGES_REQUESTED** [review]
Hi, can you rebase this onto the current master ? Other than that it seems fine, except for more documentation.

**Comment by a1phyr** [comment]
Done

**Comment by 0xpr03** [comment]
Thanks! I'll add an example to the lib file. Maybe it'll be worth wile to add a tokio feature where we also implement this for async tokio channels. Problem is though, that we'll have to select between tokio::sync::broadcast and tokio::sync::watch or tokio::sync::mpsc.

**Pull request merged**